comp/api/api/component.go

```diff
@@ -38,3 +41,25 @@ type Component interface {
 	StopServer()
 	ServerAddress() *net.TCPAddr
 }
+
+// EndpointProvider is an interface to register api endpoints
+type EndpointProvider interface {
+	http.Handler
+
+	Method() string
```

Comment:
The gorilla/mux function `Methods` accept a [variadic arguments](https://pkg.go.dev/github.com/gorilla/mux@v1.8.1#Route.Methods). A route can be register for multiple methods. `GET`, `POST`, etc.  I think it would be convenient to allow for such functionality to be supported via the `EndpointProvider`